From 16e307d1efadbc7cf10d23e55432ae183dcccd9f Mon Sep 17 00:00:00 2001
From: Malik Akbar Hashemi Rafsanjani <malikrafsan@meta.com>
Date: Mon, 14 Apr 2025 04:14:17 -0700
Subject: [PATCH] fallback to meta mode when UPN mode failed

---
 http/mutual_tls.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/http/mutual_tls.cpp b/http/mutual_tls.cpp
index d11c269..0d567e5 100644
--- a/http/mutual_tls.cpp
+++ b/http/mutual_tls.cpp
@@ -253,6 +253,12 @@ std::shared_ptr<persistent_data::UserSession> verifyMtlsUser(
     }
 
     std::string sslUser = getUsernameFromCert(peerCert);
+    if (sslUser.empty() && BMCWEB_META_TLS_COMMON_NAME_PARSING)
+    {
+        // TODO: Remove this fallback once the UPN certificate are fully rolled out
+        BMCWEB_LOG_WARNING("Fallback to Meta Inc. CommonName parsing");
+        sslUser = getMetaUserNameFromCert(peerCert);
+    }
     if (sslUser.empty())
     {
         BMCWEB_LOG_WARNING("Failed to get user from peer certificate");
-- 
2.47.1

