From 2cb325fa07a1d10f268b658d78f71a1e6e0c9ce2 Mon Sep 17 00:00:00 2001
From: Malik Akbar Hashemi Rafsanjani <malikrafsan@fb.com>
Date: Mon, 24 Feb 2025 08:36:37 -0800
Subject: [PATCH] increase the systemMaxUserNameLen

Signed-off-by: Malik Akbar Hashemi Rafsanjani <malikrafsan@fb.com>
---
 user_mgr.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index d1c525b..23ae065 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -55,7 +55,7 @@ namespace user

 static constexpr const char* passwdFileName = "/etc/passwd";
 static constexpr size_t ipmiMaxUserNameLen = 16;
-static constexpr size_t systemMaxUserNameLen = 30;
+static constexpr size_t systemMaxUserNameLen = 100;
 static constexpr const char* grpSsh = "ssh";
 static constexpr int success = 0;
 static constexpr int failure = -1;
--
2.43.5
